File: answers/leetcode/Binary Tree Right Side View/Binary Tree Right Side View.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
class Solution {
    public List<Integer> rightSideView(TreeNode root) {
        List<Integer> view = new ArrayList<Integer>();
        this.dfs(root, 0, view);
        return view;
    }
    
    private void dfs(TreeNode root, int level, List<Integer> view) {
        if (null == root) {
            return;
        }
        while (level >= view.size()) {
            view.add(-1);
        }
        view.set(level, root.val);
        this.dfs(root.left, level + 1, view);
        this.dfs(root.right, level + 1, view);
    }
}

File: answers/leetcode/Binary Tree Right Side View/Binary Tree Right Side View.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution(object):
    def rightSideView(self, root):
        """
        :type root: TreeNode
        :rtype: List[int]
        """

        right_list = []
        queue = [root]
        while len(queue):
            right = None
            for i in range(len(queue)):
                current = queue.pop(0)
                if current:
                    if None == right:
                        right = current.val
                    queue.append(current.right)
                    queue.append(current.left)
            if right:
                right_list.append(right)
        return right_list
